### ISSUE TITLE
Number of Extracted Tweets

### ISSUE THREAD

**Issue by DavidPerea** [haccer/twint#106]
### Initial Check 
I have made sure to check the following.

[] Python version is 3.6.2
[] Using the latest version of Twint.

### Command
I run this command to get the **total** of his published tweets.

`python Twint.py -u malaga -o file.csv --csv`

### Description of Issue
I hope this question is not duplicated. I have reviewed the closed questions and I have not seen one related to this topic directly.

I intend to extract all the tweets from a user. However, I can not extract the total number of tweets that the user has published. Also every time I execute the code I get a different tweets number.

Why does this happen?

Why is it that the number of extracted tweets varies?

### OS Details
I am using Windows10.

**Comment by pielco11** [comment]
Same problem here, sometimes it stops at just 10 tweets and sometimes it goes to end.

It seems to be a problem with Twitter so I suggest you to try over time, this works for me.

I don't see a specific pattern of the problem at every try so it's quite hard for me to dig further into the problem, but if you get this feel free to let us know!

**Comment by haccer** [comment]
I am seeing the issue,

I will look into this further

**Comment by haccer** [comment]
Weird,

Seems right now I cannot collect anything past 2018-05-14

**Comment by DavidPerea** [comment]
So is. 
Neither I can get more tweets from that date. I do not know why. It's strange.

**Comment by haccer** [comment]
Okay, I've just fixed this in the recent commit. I'll update the PyPI later today after I add some more stuff

**Comment by haccer** [comment]
I've tested with `python Twint.py -u malaga -o file.csv --csv` and everything seems to be running back to normal. Thanks @DavidPerea for reporting this!

**Comment by DavidPerea** [comment]
Thank you very much @haccer 

But should I make any changes?
Is that I run the command again and it happens the same. I can not get tweets from May 2018.

**Comment by haccer** [comment]
Did you update the repo?

**Comment by haccer** [comment]
I uploaded changes to pypi, so `pip install twint --upgrade` should suffice to update

**Comment by DavidPerea** [comment]
True. Now that works. 
I had not used that command to update the repo. 
Thank you very much really. You are a genius.

**Comment by DavidPerea** [comment]
Sorry to return to this topic. But I'm going to detail some more data, to see if it is possible to locate the error.

It's curiosity that when I use the command `--profile-full` I do not get all the tweets. It depends the user gets a number of different tweets, but all are around 3000 tweets. However, I try the same user at different times and always get the same number of tweets. So I think it may be something from the code, may not it?

Can you get all the tweets with the command `--profile-full`?

**Comment by haccer** [comment]
Hi @DavidPerea 

Can you provide the user so I may take a better look?

**Comment by DavidPerea** [comment]
The problem is that there are many tweets, about 5900 tweets and you will have to wait a long time. I'm embarrassed that you have to wait so long.
But, for example, is this user **_AytoSanFernando_**:

`python Twint.py -u AytoSanFernando --profile-full -o file.csv --csv`

Without `--profile-full` I get them until the end, but since I'm interested in getting the retweets I use the `--profile-ful` and in this case I only get 3175 tweets

**Comment by haccer** [comment]
Yes it is quite a long time lol @DavidPerea , I'll take a look later tonight or maybe this afternoon

**Comment by DavidPerea** [comment]
Of course, no problem. When you have time.

I tell you better another user, **_AytoHuelva_**, who has fewer tweets, 4600. However, I only get 3115 tweets.

`python Twint.py -u AytoHuelva --profile-full -o file.csv --csv`

All this happens to me since the last update of the package.

You already communicate to me whether it has been solved or not. Thank you very much!!

**Comment by Nestor75** [comment]
I faced the same issue also

**Comment by haccer** [comment]
@DavidPerea A quick thought, 

3115 and 3175 are very close, perhaps my second method for grabbing retweets is limited to around that number; I haven't confirmed this yet, I will later though. 

Question:
Does the last Tweet you grab with the `--profile-full` match the last Tweet with the regular option that utilizes Twitter's search? (i.e. their earliest Tweet)

**Comment by DavidPerea** [comment]
The strange thing is that before making the last update, I could get all the tweets with `--profile-full`. For that my doubt. So I doubt that this second method is limited to that number. I do not know, I'm a little confused.

And excuse me @haccer, you could repeat the question with other words. I do not understand it well.

**Comment by DavidPerea** [comment]
The strange thing is that before making the last update, I could get all the tweets with --profile-ful. For that my doubt. So I doubt that this second method is limited to that number. I do not know, I'm a little confused.

And excuse me @haccer , you could repeat the question with other words. I do not understand it well.

**Comment by DavidPerea** [comment]
Thinking well, you may be right @haccer . Because, I'm not sure if before with `-profile-full`, I could get all the tweets. So maybe with this second method if that limit exists.

**Comment by Nestor75** [comment]
I am running the query but is quite large so I will let you know if it works as soon as it finishes

**Comment by Nestor75** [comment]
I still face problems getting the tweets :( there are many left [https://github.com/haccer/twint/issues/141#issuecomment-394861498](url)

**Comment by haccer** [comment]
Okay I looked at this tonight.

Number of Tweets advertised on the profile: 4,662

### Test 1
- Command Ran: python3 Twint.py -u aytohuelva -o file.csv --csv
- Total Tweets Collected:  4,283
- Final Tweet:  `565119429986770944 2015-02-10 12:05:46 UTC <AytoHuelva> El jueves vive el Día de los Enamorados con el #concierto de la Banda. 20 h. Casa Colón,  #Huelva gratis  @BSMHuelva pic.twitter.com/Jp2hS4Mdt9 `

### Test 2
- Command Ran: python3 Twint.py -u aytohuelva -o file.csv --csv --profile-full
- Total Tweets Collected: 3,124
- Final Tweet: `700258959714873348 2016-02-18 05:02:00 EDT RT <detreslados> @AytoHuelva Jueves 18, 9:00PM, Gran Teatro Huelva, Delbosque, por tan solos 5€. ¡No lo dudes y ven a conocerlos!  https://www.youtube.com/watch?v=qGmXwMBNmg4 … @aytoHuelva`

### Test 3
- Command Ran: python3 Twint.py -u aytohuelva -o file.csv --csv --retweets
- Total Tweets Collected: 1,020
- Final Tweet: `932945666078277632 2017-11-21 12:15:54 UTC <AytoHuelva> Abrimos el plazo de recogida de solicitudes  para la participación en un nuevo proyecto de inserción juvenil para menores de 30 inscritos en el Fichero de Garantía Juvenil. Toda la información en el enlace:   https://goo.gl/H7TNhy  pic.twitter.com/eyZpB1hMX1`

I'm going to make some adjustments to the code tonight and tomorrow, we'll see if these numbers change

**Comment by DavidPerea** [comment]
Perfect, when you finish these adjustments, let me know and we will confirm if the numbers of the tweets change with respect to that test.

As always thank you very much.

**Comment by haccer** [comment]
An update @DavidPerea,

I suspect this issue is very similar to https://github.com/haccer/twint/issues/141, in which the reason the script is stopping early is because Twitter has hit a request limit and giving a 503 error... I'm rerunning the command with the requests being logged so I can see the last request when the script stops working. I should have more information about this tomorrow.

**Comment by DavidPerea** [comment]
Okay, it seems that a possible cause has already been found. Please, comment on the new information you have about this.
